### DISCUSSION TITLE
Why does this use the assistant API

### DISCUSSION THREAD

**Discussion by gregnwosu** [VRSEN/agency-swarm#45] [General]
couldnt the same thing be achieved via the normal openai interface with function calling?

**Comment by kendalleasterly** [comment]
my understanding is that by separating different functions into different agents, that allows for a lot more specialization and focus for each agent, and the more focused these AI's are the better

**Comment by hakimmanik** [comment]
I also had the same question, through your reply reflects using Assistant API is cause of multiple focused agents are better at doing. but same thing can also be achieved with OpenAI Chat API?

**Comment by nicko-ai** [comment]
I guess this discussion is outdated, but the general idea is to benefit from OpenAI's existing infrastructure. You can get more info in our new docs: https://agency-swarm.ai/welcome/overview

In the new framework version based on the Responses API, you’ll have full control over your conversation history and built-in support for any LLMs.
